federation/cmd/federation-controller-manager/app/options/options.go

```diff
@@ -32,6 +32,14 @@ type ControllerManagerConfiguration struct {
 	Port int `json:"port"`
 	// address is the IP address to serve on (set to 0.0.0.0 for all interfaces).
 	Address string `json:"address"`
+	// dnsProvider is the provider for dns services.
+	DnsProvider string `json:"dnsProvider"`
+	// dnsConfigFile is the path to the dns provider configuration file.
+	DnsConfigFile string `json:"ndsConfigFile"`
```

Comment:
You are right.
Being fixed in https://github.com/kubernetes/kubernetes/pull/27158

Though the json field name is not useful yet.